src/ring_theory/algebra_tower.lean

```diff
@@ -272,7 +268,7 @@ theorem smul_mem_span_smul_of_mem {s : set S} {t : set A} {k : S} (hks : k ∈ s
 span_induction hks (λ c hc, subset_span $ set.mem_smul.2 ⟨c, x, hc, hx, rfl⟩)
   (by { rw zero_smul, exact zero_mem _ })
   (λ c₁ c₂ ih₁ ih₂, by { rw add_smul, exact add_mem _ ih₁ ih₂ })
-  (λ b c hc, by { rw is_algebra_tower.smul_assoc, exact smul_mem _ _ hc })
+  (λ b c hc, by { rw is_scalar_tower.smul_assoc, exact smul_mem _ _ hc })
```

Comment:
I think we should make `submodule.smul_assoc` protected so that we won't ever need to worry about name overloading and can instead always use `smul_assoc`.
https://github.com/leanprover-community/mathlib/blob/ee7bb12ca87b7b52894fa194b847fa8680199565/src/ring_theory/ideal/operations.lean#L133
https://github.com/leanprover-community/mathlib/blob/ee7bb12ca87b7b52894fa194b847fa8680199565/src/ring_theory/ideal/operations.lean#L872

Reply:
Done.